components/style/data.rs

```diff
@@ -175,25 +175,24 @@ impl NodeData {
         self.styles = match mem::replace(&mut self.styles, Uninitialized) {
             Uninitialized => Previous(f()),
             Current(x) => Previous(Some(x)),
-            _ => panic!("Already have previous styles"),
```

Comment:
Yeah, at least for now it's more convenient for gather_previous_styles to be idempotent.